### ISSUE TITLE
Panic on OSX: D24UnormS8Uint

### ISSUE THREAD

**Issue by paulkernfeld** [Yatekii/sailor#2]
```
> cargo run --bin sailor --no-default-features --features metal                                                 paul@compy386:~/repos/sailor on master -> origin/master
    Finished dev [unoptimized + debuginfo] target(s) in 0.28s
     Running `target/debug/sailor`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Format(D24UnormS8Uint)', src/libcore/result.rs:999:5
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /Users/vsts/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /Users/vsts/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:200
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:214
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:477
   7: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:384
   8: rust_begin_unwind
             at src/libstd/panicking.rs:311
   9: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  10: core::result::unwrap_failed
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/libcore/macros.rs:18
  11: core::result::Result<T,E>::unwrap
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/libcore/result.rs:800
  12: wgpu_native::device::device_create_texture
             at /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.3.3/src/device.rs:751
  13: wgpu_device_create_texture
             at /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.3.3/src/device.rs:826
  14: wgpu::Device::create_texture
             at /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.3.0/src/lib.rs:851
  15: sailor::drawing::painter::Painter::create_stencil
             at src/bin/drawing/painter.rs:594
  16: sailor::drawing::painter::Painter::init
             at src/bin/drawing/painter.rs:158
  17: sailor::main
             at src/bin/main.rs:48
  18: std::rt::lang_start::{{closure}}
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/libstd/rt.rs:64
  19: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:49
  20: std::panicking::try::do_call
             at src/libstd/panicking.rs:296
  21: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:82
  22: std::panicking::try
             at src/libstd/panicking.rs:275
  23: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  24: std::rt::lang_start_internal
             at src/libstd/rt.rs:48
  25: std::rt::lang_start
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/libstd/rt.rs:64
  26: sailor::main
```

**Comment by Yatekii** [comment]
Working on it :) The problem atm is that Metal does not support this kind of texture format. There is a workaround and I am working to get it running asap :)

**Comment by Yatekii** [comment]
hey @paulkernfeld wgpu 0.4.0 is finally out. Do you mind checking whether it works on macOS now? I just updated the deps and made code changes required.

**Comment by paulkernfeld** [comment]
Nice, it looks like this bug has been fixed! When I run the application now, it exits with this message:

```
zsh: floating point exception  cargo run --bin sailor
```

**Comment by Yatekii** [comment]
Wait what? That sounds more like a zsh issue than a sailor issue? Do you have any idea what this could be? It seems like it does not even run now ^^

**Comment by paulkernfeld** [comment]
Yeah, it could be a zsh problem although I haven't seen it before. Is it possible that the application is throwing a `SIGFPE` or something of that nature? I have experienced that it does run for a few seconds before exiting.

**Comment by Yatekii** [comment]
Did you try running in debug mode? Because it should panic on DBZ (which is the most likely cause I guess). I think there was some DBZ but it wasn't an issue, so I just left it for starters. Some of this comes from lyon having some bugs in the tesselators which were being revamped during summer. I can check again if those are done now.

**Comment by paulkernfeld** [comment]
Same thing; here's some more information from the OSX crash reporter:

```
rocess:               sailor [5998]
Path:                  /Users/USER/*/sailor
Identifier:            sailor
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        zsh [5340]
Responsible:           iTerm2 [5331]
User ID:               501

Date/Time:             2019-12-15 12:23:55.864 -0500
OS Version:            Mac OS X 10.15.1 (19B88)
Report Version:        12
Anonymous UUID:        B39D305A-D903-FA10-AD83-11FF3D9C57B2

Sleep/Wake UUID:       9B3A8BBB-88A8-4163-A06C-E54ED6F8E5D8

Time Awake Since Boot: 25000 seconds
Time Since Wake:       46000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_ARITHMETIC (SIGFPE)
Exception Codes:       EXC_I386_DIV (divide by zero)
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Floating point exception: 8
Termination Reason:    Namespace SIGNAL, Code 0x8
Terminating Process:   exc handler [5998]
```

**Comment by Yatekii** [comment]
Hmm ... don't you get any stacktrace? :/ Normally I would get a stacktrace in debug mode. Kinda hard to find that DBZ without it ^^ I can run some tests on linux, but it's kinda funny how it seems to be a non-issue for me.

**Comment by Yatekii** [comment]
This seems to be finally resolved with wgpu 0.5.0 =) Maybe you wanna give it a try @paulkernfeld .
Now just all the other issues need fixing :/

**Comment by paulkernfeld** [comment]
It runs now on MacOS, yay! 😀
